templates/ui-ingress.yaml

```diff
@@ -0,0 +1,48 @@
+{{- if (and (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) (or (and (ne (.Values.ui.enabled | toString) "-") .Values.ui.enabled) (and (eq (.Values.ui.enabled | toString) "-") .Values.global.enabled)) (or (and (ne (.Values.ui.service.enabled | toString) "-") .Values.ui.service.enabled) (and (eq (.Values.ui.service.enabled | toString) "-") .Values.global.enabled))) }}
+{{- if (and (ne (.Values.ui.ingress.enabled | toString) "-") .Values.ui.ingress.enabled) }}
+{{- $serviceName := printf "%s-%s" (include "consul.fullname" .) "ui" -}}
+apiVersion: extensions/v1
+kind: Ingress
+metadata:
+  name: {{ template "consul.fullname" . }}-ingress
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: ui
+  {{- if .Values.ui.ingress.annotations }}
+  annotations:
+    {{ tpl .Values.ui.ingress.annotations . | nindent 4 | trim }}
+  {{- end}}
+spec:
+  rules:
+  {{ $global := .Values.global }}
+  {{- range .Values.ui.ingress.hosts }}
+  - host: {{ .host | quote }}
+    http:
+      paths:
+      {{- if (or (not $global.tls.enabled) (not $global.tls.httpsOnly)) }}
```

Comment:
The only difference I'm seeing between L27-32 and L35-49 is the servicePort.  So why not just make that the conditional instead of the entire block?

Reply:
I guess the only motivation there would be in a situation when TLS is enabled but both HTTP and HTTPs traffic is allowed, we would need both the block. I can look at DRY-ing this up though.

Reply:
I see your point from a what's allowed perspective, but not sure why someone would create an Ingress and allow port 80 traffic, if 443 is available for the UI.  I would _think_ that the primary reason for having 80 and 443 enabled on consul is because they are migrating from 80 to 443 and havent finished shoring up the clients still using 80.

Reply:
And in the case where they are both enabled how does the ingress know which backend to send to if they are defined the exact same way but with different servicePort? This is just a general curiousity question, if the path were different, I could see how it would work.  But I'd almost think this would be an invalid configuration for an Ingress.

Reply:
In case the Ingress controller does not support TLS passthrough, only supporting port 443 will not work. In the case when both ports are exposed, the controller multiplexes the requests across the ports and responds with whichever response is successful. 
That is a really good questions though. I tested it with both ports open without the passthrough and it worked like a treat.

Reply:
This allows for maximum flexibility but ensures the ability to expose just the one port (`443`) if that is so desired. Did I understand your question correctly?